plugin-server/src/utils/action-manager-cdp.ts

```diff
@@ -0,0 +1,88 @@
+import { PostgresRouter, PostgresUse } from './db/postgres'
+import { LazyLoader } from './lazy-loader'
+
+export interface Action {
+    id: number
+    name: string
+    team_id: number
+    deleted: boolean
+    bytecode: any
+    bytecode_error: string | null
+    steps_json: any
+    created_at: string
+    updated_at: string
+}
+
+export class ActionManagerCDP {
+    private lazyLoader: LazyLoader<Action[]>
+
+    constructor(private postgres: PostgresRouter) {
+        this.lazyLoader = new LazyLoader({
+            name: 'ActionManagerCDP',
+            refreshAge: 5 * 60 * 1000, // 5 minutes
+            refreshJitterMs: 60 * 1000, // 1 minute
+            loader: async (teamIds: string[]) => {
+                return await this.fetchActions(teamIds)
+            },
+        })
+    }
+
+    public async getActionsForTeam(teamId: number): Promise<Action[]> {
+        const actions = await this.lazyLoader.get(String(teamId))
+        return actions || []
+    }
+
+    public async getActionsForTeams(teamIds: number[]): Promise<Record<string, Action[] | null>> {
+        return this.lazyLoader.getMany(teamIds.map(String))
+    }
+
+    public markTeamForRefresh(teamId: number): void {
+        this.lazyLoader.markForRefresh(String(teamId))
+    }
+
+    private async fetchActions(teamIds: string[]): Promise<Record<string, Action[]>> {
```

Comment:
This is totally fine for now - later we will want to be more clever about this. We don't need to reload all actions for a team when one reloads for example. We will likely want to do it like the hog function manager - one loader for the list, a different one for the individual actions